docs/manual.md

```diff
@@ -807,91 +805,294 @@ user:
+```
+
+When a transformed test fails, it will detail the transformers used, the `-o include_raw` option can be used to include the raw, untransformed attribute value:
+```
+$ goss v
+F
+
+Failures/Skipped:
+
+KernelParam: net.core.somaxconn: value:
+Expected
+    128
+to be >
+    200
+the transform chain was
+    [{"to-numeric":{}}]
+
+Total Duration: 0.001s
+Count: 1, Failed: 1, Skipped: 0
+
+
+$ goss v -o include_raw
+F
 
-$ goss validate
-..
+Failures/Skipped:
+
+KernelParam: net.core.somaxconn: value:
+Expected
+    128
+to be >
+    200
+the transform chain was
+    [{"to-numeric":{}}]
+the raw value was
+    "128"
 
 Total Duration: 0.001s
-Count: 2, Failed: 0
+Count: 1, Failed: 1, Skipped: 0
+
 ```
 
-## Advanced Matchers
-Goss supports advanced matchers by converting json input to [gomega](https://onsi.github.io/gomega/) matchers.
+### Advanced Matchers
 
-### Examples
+Goss supports advanced matchers by converting YAML input to [gomega](https://onsi.github.io/gomega/) matchers.
+
+#### String Matchers
 
-Validate that user `nobody` has a `uid` that is less than `500` and that they are **only** a member of the `nobody` group.
+These will convert the system attribute to a string prior to matching.
 
+* `'55'` - Checks that the numeric is "55" when converted to string
+* `have-prefix: pre` - Checks if string starts with "pre"
+* `have-suffix: suf` - Checks if string ends with "suf"
+* `match-regexp: '.*'` - Checks if string matches regexp
+* `contain-substring: '2'` - Checks if string contains "2"
+
+Example:
 ```yaml
-user:
-  nobody:
-    exists: true
-    uid:
-      lt: 500
-    groups:
-      consist-of: [nobody]
+matching:
+  example:
+    content: 42
+    matches:
+      and:
+        - '42'
+        - have-prefix: '4'
+        - have-suffix: '2'
+        - match-regexp: '\d{2}'
+        - contain-substring: '2'
 ```
 
-Matchers can be nested for more complex logic, for example you can ensure that you have 3 kernel versions installed and none of them are `4.1.0`:
+#### Numeric matchers
 
+These will convert the system attribute to a numeric prior to matching.
+
+* `42` - If the expected type is a number
+* `gt, ge, lt, le` - Greater than, greater than or equal, less than, etc..
+
+Example:
 ```yaml
-package:
-  kernel:
-    installed: true
-    versions:
+matching:
+  example:
+    content: "42"
+    matches:
+      and:
+        - 42
+        - 42.0
+        - gt: 40
+        - lt: 45
+```
+
+#### Array matchers
+
+These will convert the system attribute to an array prior to matching. Strings are split on "\n"
+
+
+* `contain-element: matcher` - Checks if the array contains an element that passes the matcher
+* `contain-elements: [matcher, ...]` - checks if the array is a superset of the provided matchers
+* `[matcher, ...]` - same as above
+* `equal: [value, ...]` - Checks if the array is exactly equal to provided array
+* `consist-of: [matcher, ...]` - Checks if the array consists of the provided matchers (order does not matter)
```

Comment:
Is it to late to rename `consist-of` to `satisfies`?

Reply:
consist-of has been around since 2016. Maybe as part of 4 or a 4.X release I can add other aliases, but I prefer not breaking the old interface.

Reply:
Ah, gotcha. Yeah, definitely not worth a breaking change. If it's possible easily to alias, that might be nice, but ... not that nice.